static/app/components/forms/controls/selectOption.tsx

```diff
@@ -56,17 +56,28 @@ function SelectOption(props: Props) {
           innerWrapProps={{'data-test-id': value}}
           labelProps={{as: typeof label === 'string' ? 'p' : 'div'}}
           leadingItems={
-            <Fragment>
-              <CheckWrap isMultiple={isMultiple} isSelected={isSelected}>
-                {isSelected && (
-                  <IconCheckmark
-                    size={isMultiple ? 'xs' : 'sm'}
-                    color={isMultiple ? 'white' : undefined}
-                  />
-                )}
-              </CheckWrap>
-              {data.leadingItems}
-            </Fragment>
+            itemProps.__isNew__ ? (
```

Comment:
I’m not fond of using "__isNew__", as it doesn’t feel quite right, but I don’t have a better alternative at the moment. For what it’s worth, newer versions of react-select still use it. You can see an example here: [react-select GitHub](https://github.com/JedWatson/react-select/blob/06e34882638d1526b9f5a1238bb567a3e9460ce5/packages/react-select/src/useCreatable.ts#L98-L102).

Reply:
We could create our own prop to check if a value is new, but this would require manually updating every place we use it. Otherwise, the "create" option won’t show the plus icon. That might be fine as well

Reply:
I would rather not show plus icon, than using the `__isNew__` attribute over which we have no control, and is not a public property, but I have no strong opinion here

Reply:
the whole data prop does not have a type, but we are using props from it 🙂 the `__isNew__` feels odd only because of the `__`